### ISSUE TITLE
Share the Dockerfile of the image yue9944882/java-model-gen

### ISSUE THREAD

**Issue by yuanzac** [kubernetes-client/java#753]
We can get the docker image, yue9944882/java-model-gen, from dockerhub. But due to the network limitation, we need to modify the maven configuration of the docker image. If the Dockerfile is given, we can modify the docker images more easily.

**Comment by yue9944882** [comment]
@yuanzac it's actually right here https://github.com/yue9944882/java-model-gen, i am having troubling finding more bandwidth to write a decent user-facing documentation before the repo goes public..

**Comment by yue9944882** [comment]
hitting https://github.com/kubernetes/kubernetes/issues/81445?

**Comment by yuanzac** [comment]
@yue9944882，
The classes are generated, But some attributes of CRD are missing.
I add the "type: object" to the TFflow crd yaml like this:
`
...  
validation:
    openAPIV3Schema:
      type: object
      properties:
        spec:
          type: object
          properties:
            tfReplicaSpecs:
              type: object
              properties:
                # The validation works when the configuration contains
                # `Worker`, `PS` or `Chief`. Otherwise it will not be validated.
                Worker:
                  type: object
                  properties:
                    replicas:
                      type: integer
                      minimum: 1
                PS:
                  type: object
                  properties:
                    replicas:
                      type: integer
                      minimum: 1
                Chief:
                  type: object
                  properties:
                    replicas:
                      type: integer
                      minimum: 1
                      maximum: 1` 

But the TFjob creation yaml is like this:
`
...
spec:
  tfReplicaSpecs:
    PS:
      replicas: 2
      restartPolicy: Never
      template:
        spec:
          containers:
            - name: tensorflow
              image: kubeflow/tf-dist-mnist-test:1.0
    Worker:
      replicas: 4
      restartPolicy: Never
      template:
        spec:
          containers:
            - name: tensorflow
              image: kubeflow/tf-dist-mnist-test:1.0
` 
As it contains attributes like template and restartPolicy that CRD has not defined. The creation will get the following error:
`error: error validating "./tf_job_mnist.yaml": error validating data: [ValidationError(TFJob.spec.tfReplicaSpecs.PS): unknown field "restartPolicy" in org.kubeflow.v1.TFJob.spec.tfReplicaSpecs.PS, ValidationError(TFJob.spec.tfReplicaSpecs.PS): unknown field "template" in org.kubeflow.v1.TFJob.spec.tfReplicaSpecs.PS, ValidationError(TFJob.spec.tfReplicaSpecs.Worker): unknown field "restartPolicy" in org.kubeflow.v1.TFJob.spec.tfReplicaSpecs.Worker, ValidationError(TFJob.spec.tfReplicaSpecs.Worker): unknown field "template" in org.kubeflow.v1.TFJob.spec.tfReplicaSpecs.Worker]; if you choose to ignore these errors, turn validation off with --validate=false
`
The TFjob would be created if we deleted "type: object" from the CRD yaml, which would block java code generation from openapi.
Could you share some comments or clues? 

Thanks a lot~

**Comment by narayanan** [comment]
@yuanzac @yue9944882 Any update on this? I'm running into the same issue, I only see `xxxList.java` & `xxxMetadata.java` files getting generated. What am I missing? How do I get the other files generated?

**Comment by DaspawnW** [comment]
I have the same issue like @narayanan. Only ListMetadata.java and List.java are generated

**Comment by narayanan** [comment]
@DaspawnW I got the tool to generate code fine.. Code generator is very sensitive about the structure / format of the Open API schema. In my case, the indentation / spacing on one array was incorrect, once I fixed it, code generator generated code for all classes. However I had to spend couple of hours on trial-and-error basis to find out the root cause. I wish the code generator printed some error.
